File: src/app/core/services/grade.service.ts
import { Observable } from 'rxjs';
import { GradeSelect } from '../models/grade.liste';
import { RestResponse } from '../models/rest.response';

export interface GradeService {
  findAllSelect(libelle:string): Observable<RestResponse<GradeSelect[]>> ;
  findAll(page:number): Observable<RestResponse<GradeSelect[]>> ;
  create(gradeSelect:GradeSelect): Observable<RestResponse<GradeSelect>> ;
}

File: src/main/java/com/rest/gestSchool/gestSchoolRest/services/GradeService.java
package com.rest.gestSchool.gestSchoolRest.services;


import com.rest.gestSchool.gestSchoolRest.web.dto.response.grade.GradeSelectDto;
import com.spring_GestSchool.spring_GestSchool.data.entities.Grade;
import com.spring_GestSchool.spring_GestSchool.data.entities.Spécialite;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface GradeService {
    List<Grade> getAllGradeByLibelle(String libelle);

    Page<Grade> getAllByPaginate(Pageable pageable);

    Grade create(GradeSelectDto gradeSelectDto);
}
